### ISSUE TITLE
Build a pile of Cubes” for C#,  causes stack overflow

### ISSUE THREAD

**Issue by TommieEdwardBerry** [Codewars/codewars.com#756]
This bug is about a specific KATA, “Build a pile of Cubes” for C#.
But it crashes the system because the recursion depth is too deep for your setup.
I want to find out why this limit is in there and can I increase it?
I attached the program, picture of the passing KATA examples and stack overflow dump.
I hope this is the right place to submit.
[Build_a_pile_of_Cubes_in_Csharp.txt](https://github.com/Codewars/codewars.com/files/681471/Build_a_pile_of_Cubes_in_Csharp.txt)
<img width="1421" alt="passingkataexamples" src="https://cloud.githubusercontent.com/assets/24882862/21597273/6db66474-d10d-11e6-8c4f-a3e40c06c0cf.png">
[MyError.txt](https://github.com/Codewars/codewars.com/files/681475/MyError.txt)

**Comment by jhoffner** [comment]
I'm not really sure what the issue is based off of the screenshot but no the stack depth can't be changed. I would file an issue on the kata discourse page as github is only for sitewide issues.

**Comment by g964** [comment]
Nobody said that recursion was mandatory.

**Comment by TommieEdwardBerry** [comment]
I rewrote it without recursion, but the recursion version is way simpler and it is a shame that the system stack is that limited in a time when I carry more memory than that in my pocket. It should be increased.
Just saying,
Tommie

From iPhone6+

**Comment by jhoffner** [comment]
It's running a basic mono executable on a machine with 1gb of ram. If there are any severe memory limits this is the first I'm hearing. Are you sure the memory footprint isn't larger than you think, or that you are in an infinite loop?

**Comment by TommieEdwardBerry** [comment]
I am not sure, I could run all test cases on my pc, under Visual Studio. I will go to my working test case and get all the tests and try it and report back. I may even be able to get the stack size used. Stand by, I will report back to answer your question.
Tommie

From iPhone6+

**Comment by TommieEdwardBerry** [comment]
OK, Nothing wrong with your site or the program.

The example recursions were very low counts, but the full test uses recursions greater that 50,000 !!

I moved it all to visual studio on my PC and I get the same Stack Overflow on my PC.

My PC can do 6739 recursions and fails at 6740 , not even close to the 50,000 recursions required to get these answers.

I guess 50,000 recursions is too much.

Sorry for the fire, this issue is closed.

Thanks for helping,

Tommie

 

PS: I attached the Stack OverFlow exception data if you are interested.
